tests/dogtag/pytest-ansible/pytest/performance_test/test_acme_cert_enrollment.py

```diff
@@ -0,0 +1,235 @@
+    order, _, order_headers = send_signed_request(account_key, directory['newOrder'], order_payload,
+                                                  "Error creating new order", acct_headers=acct_headers)
+    log.info("Order created!")
+
+    # get the authorizations that need to be completed
+    for auth_url in order['authorizations']:
+        authorization, _, _ = send_signed_request(account_key, auth_url, None, "Error getting challenges",
+                                                  acct_headers=acct_headers)
+        domain = authorization['identifier']['value']
+        log.info("Verifying {0}...".format(domain))
+
+        # find the http-01 challenge and write the challenge file
+        challenge = [c for c in authorization['challenges'] if c['type'] == "http-01"][0]
+        token = re.sub(r"[^A-Za-z0-9_\-]", "_", challenge['token'])
+        keyauthorization = "{0}.{1}".format(token, thumbprint)
+        wellknown_path = os.path.join(acme_dir, token)
+        with open(wellknown_path, "w") as wellknown_file:
+            wellknown_file.write(keyauthorization)
+
+        # say the challenge is done
+        send_signed_request(account_key, challenge['url'], {}, "Error submitting challenges: {0}".format(domain),
+                            acct_headers=acct_headers)
+        authorization = poll_until_not(account_key, auth_url, ["pending"],
+                                       "Error checking challenge status for {0}".format(domain),
+                                       acct_headers=acct_headers)
+        if authorization['status'] != "valid":
+            raise ValueError("Challenge did not pass for {0}: {1}".format(domain, authorization))
+        os.remove(wellknown_path)
+        log.info("{0} verified!".format(domain))
+
+    # finalize the order with the csr
+    log.info("Signing certificate...")
+    csr_der = cmd(["openssl", "req", "-in", csr, "-outform", "DER"], err_msg="DER Export Error")
+    send_signed_request(account_key, order['finalize'], {"csr": b64(csr_der)}, "Error finalizing order",
+                        acct_headers=acct_headers)
+
+    # poll the order to monitor when it's done
+    order = poll_until_not(account_key, order_headers['Location'], ["pending", "processing"],
+                           "Error checking order status", acct_headers=acct_headers)
+    if order['status'] != "valid":
+        raise ValueError("Order failed: {0}".format(order))
+
+    # download the certificate
+    certificate_pem, _, _ = send_signed_request(account_key, order['certificate'], None, "Certificate download failed",
+                                                acct_headers=acct_headers)
+    log.info("Certificate signed!")
+    return certificate_pem
+
+
+def main(domain_name, number_of_tests_per_thread):
+    for i in range(number_of_tests_per_thread):
+        signed_crt = get_crt(domain_name, args.acme_dir, log=LOGGER)
+        sys.stdout.write(signed_crt)
```

Comment:
Here we need to get a new nonce for each thread to avoid conflicts with other threads:

```
nonce = do_request(directory['newNonce'])[2]['Replay-Nonce']
for i in range(number_of_tests_per_thread):
    signed_crt = get_crt(..., nonce=nonce)
```

Ideally the initial nonce for each thread should be obtained before `start = timer()` as well, but this is only done once so it shouldn't affect the throughput by much if we do sufficiently large number of tests per thread.